File: other/SWEX/1979.cpp
#include <stdio.h>
#define rep(i,a,b) for(int i=a;i<b;i++)

void PS(int TCnum);

int main() {
    int T;
    scanf("%d", &T);
    rep(i,1,T+1) PS(i);
    
    return 0;
}

void PS(int TCnum) {
    int N, K;
    scanf("%d %d", &N, &K);
    int kwindow = (1<<(K+1))-2; // ...001110
    int owindow = (1<<(K+2))-1; // ...011111

    // 각 줄의 값을 비트마스크로 저장
    int garo[15] = {0}; 
    int sero[15] = {0};

    rep(y,0,N) {
      rep(x,0,N) {
        int tmp;
        scanf("%d", &tmp);
        garo[y] += tmp<<(x+1);
        sero[x] += tmp<<(y+1);
      }
    }

    // 가로, 세로에서 정확히 K개의 연속된 1이 있는지 확인
    int ans = 0;
    rep(i,0,N) {
      while(garo[i]) {
        if((garo[i]&owindow) == kwindow) ans++;
        garo[i] >>= 1;
      }
      while(sero[i]) {
        if((sero[i]&owindow) == kwindow) ans++;
        sero[i] >>= 1;
      }
    }

    printf("#%d %d\n", TCnum, ans);
}

File: other/SWEX/Day_01/1979.py
import sys
sys.stdin = open('input.txt', 'r')
T = int(input())
for tc in range(1,T+1):
    N, K = map(int, input().split())
    
    kwindow = (1 << (K + 1)) - 2
    owindow = (1 << (K + 2)) - 1
    
    garo = [0] * N
    sero = [0] * N
    
    # 비트 생성
    for y in range(N):
        data = list(map(int, input().split()))
        for x in range(N):
            tmp = int(data[x])
            garo[y] |= (tmp << (x + 1))
            sero[x] |= (tmp << (y + 1))
    
    ans = 0
    
    # 가로, 세로 비트 검사
    for i in range(N):
        val_g = garo[i]
        while val_g:
            if (val_g & owindow) == kwindow:
                ans += 1
            val_g >>= 1
            
        val_s = sero[i]
        while val_s:
            if (val_s & owindow) == kwindow:
                ans += 1
            val_s >>= 1
            
    print(f"#{tc} {ans}")